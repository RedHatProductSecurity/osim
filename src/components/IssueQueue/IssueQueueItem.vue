<script setup lang="ts">
import { computed } from 'vue';

import type { FilteredIssue } from './IssueQueue.vue';

const { issue, showLabels } = defineProps<{
  issue: FilteredIssue;
  showLabels: boolean;
}>();

// Temporarily hiding 'Source' column to avoid displaying incorrect information.
// TODO: unhide it once final issue sources are defined. [OSIDB-2424]
//       and update the CSS for the column width in IssueQueue

const nonIdFields: Exclude<keyof FilteredIssue, 'id'>[] =
  ['impact', 'formattedDate', 'title', 'workflowState', 'owner'];

const hasBadges = computed(() => issue.embargoed || (!!issue.labels?.length && showLabels));
const sortedLabels = computed(() => issue.labels?.toSorted((a, b) => {
  if (
    (a.state === 'REQ' && b.state !== 'REQ')
    || (a.contributor && !b.contributor)
  ) {
    return -1;
  }
  if (
    (a.state !== 'REQ' && b.state === 'REQ')
    || (!a.contributor && b.contributor)
  ) {
    return 1;
  }

  return a.label.localeCompare(b.label);
}) ?? []);

const labelColorMap: { [key: string]: string } = {
  'ansible': '#FF9AFC',
  'community': '#BD9EA3',
  'devel-tools': '#FFFF9E',
  'kernel': '#72CBD7',
  'management': '#78E2B7',
  'middleware': '#FFAC74',
  'platforms': '#C39BFF',
  'special-handling': '#D8FF89',
  'cloud': '#A1CBF5',
  'openshift': '#FFD574',
  'other': '#86C770',
  'rhel': '#FF7B7B',
  'services': '#8292FF',
};

function getLabelColor(label: string, type: string): string {
  if (type === 'context_based') {
    return '#B9BCCB';
  }
  return labelColorMap[label] || '#FFFFFF';
}
</script>

<template>
  <tr
    class="osim-issue-queue-item"
    :class="$attrs.class"
  >
    <td
      class="osim-issue-title"
      :class="{ 'pb-0': hasBadges }"
    >
      <RouterLink :to="{ name: 'flaw-details', params: { id: issue.id } }">
        {{ issue.id }}
      </RouterLink>
    </td>
    <td
      v-for="field in nonIdFields"
      :key="field"
      :class="{ 'pb-0': hasBadges }"
    >
      {{ issue[field] }}
    </td>
    <!--<td>{{ issue.assigned }}</td>-->
  </tr>
  <tr
    v-if="hasBadges"
    class="osim-badge-lane"
    :class="$attrs.class"
  >
    <td colspan="1">
      <div class="gap-1 d-flex flex-wrap">
        <span
          v-if="issue.embargoed"
          class="badge rounded-pill text-bg-danger border border-primary"
        >Embargoed</span>
        <template v-if="showLabels">
          <template
            v-for="label in sortedLabels"
            :key="label.label"
          >
            <span
              v-if="!label.contributor"
              :style="{ backgroundColor: getLabelColor(label.label, label.type) }"
              class="badge rounded-pill border"
              :class="{
                'text-bg-warning fw-bold border-warning': label.state == 'REQ',
                'text-black': label.state != 'REQ' && label.relevant,
                'text-decoration-line-through text-bg-gray border-secondary': !label.relevant,
              }"
              :title="label.state === 'REQ' ? 'Requested' : ''"
            >{{ label.label }}</span>
          </template>
        </template>
      </div>
    </td>
    <td colspan="90%"></td>
  </tr>
</template>

<style lang="scss" scoped>
@import '@/scss/bootstrap-overrides';

td.osim-issue-title {
  max-width: 10rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

tr td {
  border: 0;
}

tr.osim-shaded td {
  background-color: $light-gray;
}

tr:hover {
  td {
    background-color: #ffe3d9;
  }

  &.osim-issue-queue-item + tr.osim-badge-lane td {
    background-color: #ffe3d9;
  }
}

tr.osim-issue-queue-item:has(+ tr.osim-badge-lane:hover) td {
  background-color: #ffe3d9;
}

.osim-badge-lane {
  .osim-show-all-labels {
    cursor: pointer;
  }
}
</style>
