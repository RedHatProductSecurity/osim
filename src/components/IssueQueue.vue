<script setup lang="ts">
import { computed, onMounted, reactive, ref, onUnmounted, watch } from 'vue';
import { DateTime } from 'luxon';
import IssueQueueItem from '@/components/IssueQueueItem.vue';
import LabelCheckbox from './widgets/LabelCheckbox.vue';
import { useUserStore } from '@/stores/UserStore';
import { FlawClassificationStateEnum } from '../generated-client';
const { userName } = useUserStore();

const emit = defineEmits(['flaws:fetch', 'flaws:load-more']);

type FilteredIssue = {
  issue: any;
  selected: boolean;
};

type ColumnField = 'id' | 'impact' | 'source' | 'created_dt' | 'title' | 'state' | 'owner';

const props = defineProps<{
  issues: any[];
  isLoading: boolean;
  isFinalPageFetched: boolean;
}>();

const issues = computed<any[]>(() => props.issues.map(relevantFields));
const issueFilter = ref('');
const selectedSortField = ref<ColumnField>('created_dt');
const isSortedByAscending = ref(false);
const isMyIssuesSelected = ref(false);
const isOpenIssuesSelected = ref(false);
const tableContainerEl = ref<HTMLElement | null>(null);

const filteredStates = computed(() => {
  const allStates = Object.values(FlawClassificationStateEnum);
  return allStates
    .filter(
      (state) =>
        state !== FlawClassificationStateEnum.Done &&
        state !== FlawClassificationStateEnum.Rejected,
    )
    .join(',');
});

const filters = computed(() => {
  const filterObj: Record<string, any> = {};

  if (isMyIssuesSelected.value) {
    filterObj.owner = userName;
  }

  if (isOpenIssuesSelected.value) {
    filterObj.workflow_state = filteredStates.value;
  }

  return filterObj;
});

const columnsFieldsMap: Record<string, ColumnField> = {
  ID: 'id',
  Impact: 'impact',
  Source: 'source',
  Created: 'created_dt',
  Title: 'title',
  State: 'state',
  Owner: 'owner',
};

const relevantIssues = computed<FilteredIssue[]>(() => {
  const filterCaseInsensitive = issueFilter.value.toLowerCase();
  const wontUseFilter = issueFilter.value.length === 0;
  return issues.value
    .filter(
      (issue: any) =>
        wontUseFilter ||
        [issue.title, issue.cve_id, issue.state, issue.source].some(
          (text) => text && text.toLowerCase().includes(filterCaseInsensitive),
        ),
    )
    .sort((a: any, b: any) =>
      isSortedByAscending.value
        ? a[selectedSortField.value].localeCompare(b[selectedSortField.value])
        : b[selectedSortField.value].localeCompare(a[selectedSortField.value]),
    )
    .map((issue) => reactive({ issue, selected: false }));
});

function updateSelectAll(selectedAll: boolean) {
  for (const filteredIssue of relevantIssues.value) {
    filteredIssue.selected = selectedAll;
  }
}

function selectSortField(field: ColumnField) {
  isSortedByAscending.value =
    selectedSortField.value === field ? !isSortedByAscending.value : false;
  selectedSortField.value = field;
}

const isSelectAllIndeterminate = computed(() => {
  if (relevantIssues.value.length === 0) {
    return false;
  }
  return relevantIssues.value.some((it) => it.selected !== relevantIssues.value[0].selected);
});

const isSelectAllChecked = computed(() => {
  return relevantIssues.value.every((it) => it.selected);
});

function relevantFields(issue: any) {
  return {
    id: issue.cve_id || issue.uuid,
    impact: issue.impact,
    source: issue.source,
    created_dt: issue.created_dt,
    title: issue.title,
    state: issue.state,
    unembargo_dt: issue.unembargo_dt,
    embargoed: issue.embargoed,
    owner: issue.owner,
    formattedDate: DateTime.fromISO(issue.created_dt).toFormat('yyyy-MM-dd'),
  };
}

onMounted(() => {
  tableContainerEl.value?.addEventListener('scroll', handleScroll); // AutoScroll Option
  emit('flaws:fetch', filters);
});

onUnmounted(() => {
  tableContainerEl.value?.removeEventListener('scroll', handleScroll);
});

// AutoScroll Method. Maybe have this as a user configurable option in the future?
function handleScroll() {
  if (props.isLoading || !tableContainerEl.value) return; // Do not load more if already loading

  const totalHeight = tableContainerEl.value.scrollHeight;
  const scrollPosition = tableContainerEl.value.scrollTop + tableContainerEl.value.clientHeight;

  if (scrollPosition >= totalHeight) {
    emitLoadMore();
  }
}

function emitLoadMore() {
  emit('flaws:load-more', filters);
}

watch(filters, () => {
  emit('flaws:fetch', filters);
});
</script>

<template>
  <div class="osim-content container osim-issue-queue">
    <div class="osim-incident-filter">
      <label>
        <!--Filter By-->
        <!--<select>-->
        <!--  <option value="Issues assigned to Me">Issues assigned to Me</option>-->
        <!-- <option value="Issues assigned to team but unowned">
          Issues assigned to team but unowned
        </option> -->
        <!--  <option value="Oldest">Oldest</option>-->
        <!--  <option value="Newest">Newest</option>-->
        <!--</select>-->

        <input
          v-model="issueFilter"
          type="text"
          class="form-text form-control"
          placeholder="Filter Issues/Flaws"
        />
      </label>
      <LabelCheckbox v-model="isMyIssuesSelected" label="My Issues" class="d-inline-block" />
      <LabelCheckbox v-model="isOpenIssuesSelected" label="Open Issues" class="d-inline-block" />

      <span
        v-if="isLoading"
        class="spinner-border spinner-border-sm d-inline-block ms-3"
        role="status"
      >
        <span class="visually-hidden">Loading...</span>
      </span>
      <span v-if="isLoading"> Loading Flaws&hellip; </span>
    </div>
    <div ref="tableContainerEl" class="osim-incident-list">
      <table class="table align-middle" :class="{ 'osim-table-loading': isLoading }">
        <thead class="sticky-top">
          <!-- <thead class=""> -->
          <tr>
            <th>
              <input
                type="checkbox"
                class="form-check-input"
                :indeterminate="isSelectAllIndeterminate"
                :checked="isSelectAllChecked"
                aria-label="Select All Issues in Table"
                @change="updateSelectAll(($event.target as HTMLInputElement).checked)"
              />
            </th>
            <th
              v-for="(field, columnName) in columnsFieldsMap"
              :key="columnName"
              :class="{ 'osim-issue-queue-table-created': columnName === 'Created' }"
              @click="selectSortField(field)"
            >
              {{ columnName }}
              <i
                :class="{
                  'opacity-0': selectedSortField !== field,
                  'bi-caret-up-fill': isSortedByAscending,
                  'bi-caret-down-fill': !isSortedByAscending,
                }"
                class="bi"
              >
              </i>
            </th>
          </tr>
        </thead>
        <tbody class="table-group-divider">
          <template v-for="(relevantIssue, index) of relevantIssues" :key="relevantIssue.id">
            <IssueQueueItem
              v-model:selected="relevantIssue.selected"
              :issue="relevantIssue.issue"
              :class="{
                'osim-shaded': index % 2 === 0,
              }"
            />
          </template>
        </tbody>
      </table>
      <button
        v-if="!isFinalPageFetched"
        class="btn btn-primary ms-4"
        type="button"
        :disabled="isLoading"
        @click="emitLoadMore"
      >
        <span
          v-if="isLoading"
          class="spinner-border spinner-border-sm d-inline-block"
          role="status"
        >
          <span class="visually-hidden">Loading...</span>
        </span>
        <span v-if="isLoading"> Loading Flaws&hellip; </span>
        <span v-else> Load More Flaws </span>
      </button>
      <p v-if="relevantIssues.length === 0">No results.</p>
      <span v-if="isFinalPageFetched" role="status">Nothing else to load.</span>
    </div>
  </div>
</template>

<style scoped lang="scss">
@import '@/scss/redhat-brand-colors';

.osim-issue-queue {
  font-family: 'Red Hat Mono', monospace;

  // tbody

  div.osim-incident-list {
    display: block;
    max-height: calc(100vh - 164px);
    overflow-y: scroll;

    &:hover::-webkit-scrollbar-thumb {
      background-color: $redhat-red-50;
    }
  }

  &:hover i {
    color: $redhat-red-50;
  }

  table {
    &.osim-table-loading {
      opacity: 0.5;
    }

    th {
      cursor: pointer;
      user-select: none;
    }

    tr td,
    tr th {
      &:not(:first-child) {
        min-width: 12ch;
      }
    }

    tbody:hover::-webkit-scrollbar-thumb {
      background-color: $redhat-red-50;
    }
  }
}
</style>
