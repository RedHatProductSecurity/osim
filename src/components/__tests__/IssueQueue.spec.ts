import { describe, it, expect, vi } from 'vitest';
import { Settings } from 'luxon';
import { flushPromises } from '@vue/test-utils';

import IssueQueueItem from '@/components/IssueQueue/IssueQueueItem.vue';
import IssueQueue from '@/components/IssueQueue/IssueQueue.vue';

import { mountWithConfig } from '@/__tests__/helpers';
import LabelCheckbox from '@/widgets/LabelCheckbox/LabelCheckbox.vue';
import type { ZodFlawType } from '@/types';
import { useSettingsStore } from '@/stores/SettingsStore';

vi.mock('@/stores/UserStore', () => ({
  useUserStore: () => ({
    jiraUsername: 'skynet',
  }),
}));

describe('issueQueue', () => {
  const mockData: Partial<ZodFlawType>[] = [
    {
      uuid: '709e9ea1-ed0f-49b8-a7ad-9164d4034849',
      cve_id: 'CVE-2903-0092',
      impact: 'MODERATE',
      title: 'title',
      unembargo_dt: '2024-02-01T00:00:00Z',
      source: 'XEN',
      embargoed: false,
      created_dt: '2021-07-29T14:50:50Z',
      updated_dt: '2024-03-15T08:53:06Z',
      classification: {
        workflow: 'DEFAULT',
        state: 'NEW',
      },
      owner: 'test@redhat.com',
    },
  ];

  const mountIssueQueue = (props?: Partial<InstanceType<typeof IssueQueue>['$props']>) => mountWithConfig(IssueQueue, {
    props: {
      issues: mockData,
      isLoading: false,
      isFinalPageFetched: false,
      total: 10,
      ...props,
    },
  });

  afterAll(() => {
    vi.clearAllMocks();
    Settings.defaultZone = 'local';
  });

  it('should fetch data from API', async () => {
    const wrapper = mountIssueQueue();

    const fetchEvents = wrapper.emitted();
    const issues = wrapper.findAllComponents(IssueQueueItem);

    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch']).toEqual([
      [
        expect.objectContaining({
          _value: { order: '-created_dt' },
        }),
      ],
    ]);
    expect(issues.length).toBe(1);
  });

  it('fetch data from API with specified parameters on MyFlaws', async () => {
    const wrapper = mountIssueQueue();

    const myIssuesCheckbox = wrapper.findAllComponents(LabelCheckbox)[0];
    const myIssuesCheckboxEl = myIssuesCheckbox.find('input[type="checkbox"]');
    await myIssuesCheckboxEl.setValue(true);
    await wrapper.vm.$nextTick();

    const fetchEvents = wrapper.emitted();
    const issues = wrapper.findAllComponents(IssueQueueItem);

    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch'][0]).toEqual([
      expect.objectContaining({
        _value: { order: '-created_dt', owner: 'skynet' },
      }),
    ]);
    expect(issues.length).toBe(1);
  });

  it('fetch data from API with specified parameters on OpenFlaws', async () => {
    const wrapper = mountIssueQueue({ issues: [] });

    const openIssuesCheckox = wrapper.findAllComponents(LabelCheckbox)[1];
    const openIssuesCheckboxEl = openIssuesCheckox.find('input[type="checkbox"]');
    await openIssuesCheckboxEl.setValue(true);
    await wrapper.vm.$nextTick();

    const fetchEvents = wrapper.emitted();
    const issues = wrapper.findAllComponents(IssueQueueItem);
    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch'][0]).toEqual([
      expect.objectContaining({
        _value: { order: '-created_dt', workflow_state: 'NEW,TRIAGE,PRE_SECONDARY_ASSESSMENT,SECONDARY_ASSESSMENT' },
      }),
    ]);
    expect(issues.length).toBe(0);
  });

  it('fetch data from API with specified parameters on sort', async () => {
    const wrapper = mountIssueQueue();

    await wrapper.findAll('th').at(0)!.trigger('click');
    const fetchEvents = wrapper.emitted();

    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch'][0]).toEqual([
      expect.objectContaining({
        _value: { order: '-cve_id,-uuid' },
      }),
    ]);
  });

  it('changes sort order on click', async () => {
    const wrapper = mountIssueQueue();

    await wrapper.findAll('th').at(0)!.trigger('click');
    await wrapper.findAll('th').at(0)!.trigger('click');
    const fetchEvents = wrapper.emitted();

    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch'][0]).toEqual([
      expect.objectContaining({
        _value: { order: 'cve_id,uuid' },
      }),
    ]);
  });

  it('removes sort order on third click', async () => {
    const wrapper = mountIssueQueue();

    await wrapper.findAll('th').at(0)!.trigger('click');
    await wrapper.findAll('th').at(0)!.trigger('click');
    await wrapper.findAll('th').at(0)!.trigger('click');
    const fetchEvents = wrapper.emitted();

    expect(fetchEvents).toHaveProperty('flaws:fetch');
    expect(fetchEvents['flaws:fetch'][0]).toEqual([
      expect.objectContaining({
        _value: {},
      }),
    ]);
  });

  it('shouldn\'t render total count when no issues', async () => {
    const wrapper = mountIssueQueue({ issues: [] });

    const filterEl = wrapper.find('div.osim-incident-filter');
    expect(filterEl.exists()).toBeTruthy();
    const countEL = filterEl.find('span.float-end');
    expect(countEL.exists()).toBeFalsy();
  });

  it('should render total count', async () => {
    const wrapper = mountIssueQueue({
      issues: Array.from({ length: 50 }).fill(mockData[0]) as ZodFlawType[],
      total: 100,
    });
    const filterEl = wrapper.find('div.osim-incident-filter');
    expect(filterEl.exists()).toBeTruthy();
    const countEL = filterEl.find('span.float-end');
    expect(countEL.exists()).toBeTruthy();
    expect(countEL.text()).toBe('Loaded 50 of 100');
  });

  it('should render loader when loading flaws', async () => {
    const wrapper = mountIssueQueue({
      issues: Array.from({ length: 50 }).fill(mockData[0]) as ZodFlawType[],
      isLoading: true,
      total: 100,
    });
    const filterEl = wrapper.find('div.osim-incident-filter');
    expect(filterEl.exists()).toBeTruthy();
    const spinner = filterEl.find('div.float-end span.spinner-border');
    expect(spinner.exists()).toBeTruthy();
  });

  it('should render create_dt in UTC format', async () => {
    Settings.defaultZone = 'Europe/Madrid';
    const wrapper = mountIssueQueue({
      issues: [{ ...mockData[0], created_dt: '2021-07-29T22:50:50Z' } as ZodFlawType],
    });

    const dateEl = wrapper.findAll('td')[2];
    expect(dateEl.exists()).toBeTruthy();
    expect(dateEl.text()).toBe('2021-07-29');
  });

  it('should render flaw labels', () => {
    const wrapper = mountIssueQueue({
      issues: [{
        ...mockData[0],
        labels: [
          { label: 'test', state: 'NEW', contributor: '' },
          { label: 'test-2', state: 'NEW', contributor: '' },
          { label: 'test-3', state: 'REQ', contributor: '' },
        ],
      } as ZodFlawType],
    });

    const labels = wrapper.findComponent(IssueQueueItem).findAll('span.badge');

    expect(labels.length).toBe(3);
    expect(labels[0].text()).toBe('test-3');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not render flaw labels that are already assigned', () => {
    const wrapper = mountIssueQueue({
      issues: [{
        ...mockData[0],
        labels: [
          { label: 'test', state: 'REQ', contributor: 'test' },
          { label: 'test-2', state: 'NEW', contributor: 'test-2' },
        ],
      } as ZodFlawType],
    });

    const labels = wrapper.findComponent(IssueQueueItem).findAll('span.badge');

    expect(labels.length).toBe(0);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not render flaw labels when isHidingLabels is true', async () => {
    const wrapper = mountIssueQueue({
      issues: [{
        ...mockData[0],
        labels: Array.from({ length: 10 }).map((_, i) => ({ label: `test-${i}`, state: 'NEW', contributor: '' })),
      } as ZodFlawType],
    });

    const store = useSettingsStore();
    store.settings.isHidingLabels = true;
    await flushPromises();

    const labels = wrapper.findComponent(IssueQueueItem).findAll('span.badge');
    expect(labels.length).toBe(0);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render flaw labels when isHidingLabels is toggled', async () => {
    const wrapper = mountIssueQueue({
      issues: [{
        ...mockData[0],
        labels: Array.from({ length: 10 }).map((_, i) => ({ label: `test-${i}`, state: 'NEW', contributor: '' })),
      } as ZodFlawType],
    });
    const store = useSettingsStore();
    store.settings.isHidingLabels = true;
    await flushPromises();

    const toggleBtn = wrapper.findAllComponents(LabelCheckbox)[2].find('input[type="checkbox"]');
    await toggleBtn.setValue(false);

    const labels = wrapper.findComponent(IssueQueueItem).findAll('span.badge');
    expect(labels.length).toBe(10);
  });
});
