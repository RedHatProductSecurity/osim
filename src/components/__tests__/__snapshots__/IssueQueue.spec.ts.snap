// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`issueQueue > should not render flaw labels that are already assigned 1`] = `
"<div data-v-48717bcb="" class="osim-content container-fluid osim-issue-queue">
  <div data-v-48717bcb="" class="osim-incident-filter">
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">My Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Open Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Hide labels</span></label></div>
    <!--v-if--><span data-v-48717bcb="" class="float-end"> Loaded 1 of 10</span>
  </div>
  <div data-v-48717bcb="" class="osim-incident-list">
    <table data-v-48717bcb="" class="table align-middle">
      <thead data-v-48717bcb="" class="sticky-top">
        <!-- <thead class=""> -->
        <tr data-v-48717bcb="">
          <th data-v-48717bcb="">ID <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Impact <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Created <i data-v-48717bcb="" class="bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Title <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">State <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Owner <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
        </tr>
      </thead>
      <tbody data-v-48717bcb="" class="table-group-divider">
        <tr data-v-15d9c71d="" class="osim-issue-queue-item osim-shaded">
          <td data-v-15d9c71d="" class="osim-issue-title pb-0"><a data-v-15d9c71d="" href="/flaws/CVE-2903-0092" class="">CVE-2903-0092</a></td>
          <td data-v-15d9c71d="" class="pb-0">MODERATE</td>
          <td data-v-15d9c71d="" class="pb-0">2021-07-29</td>
          <td data-v-15d9c71d="" class="pb-0">title</td>
          <td data-v-15d9c71d="" class="pb-0">NEW</td>
          <td data-v-15d9c71d="" class="pb-0">test@redhat.com</td>
          <!--<td>{{ issue.assigned }}</td>-->
        </tr>
        <tr data-v-15d9c71d="" class="osim-badge-lane osim-shaded">
          <td data-v-15d9c71d="" colspan="1">
            <div data-v-15d9c71d="" class="gap-1 d-flex flex-wrap">
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </div>
          </td>
          <td data-v-15d9c71d="" colspan="90%"></td>
        </tr>
      </tbody>
    </table><button data-v-48717bcb="" class="btn btn-primary" type="button">
      <!--v-if--><span data-v-48717bcb=""> Load More Flaws </span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`issueQueue > should not render flaw labels when isHidingLabels is true 1`] = `
"<div data-v-48717bcb="" class="osim-content container-fluid osim-issue-queue">
  <div data-v-48717bcb="" class="osim-incident-filter">
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">My Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Open Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox" checked=""><span data-v-b2ceeac4="" class="form-check-label">Hide labels</span></label></div>
    <!--v-if--><span data-v-48717bcb="" class="float-end"> Loaded 1 of 10</span>
  </div>
  <div data-v-48717bcb="" class="osim-incident-list">
    <table data-v-48717bcb="" class="table align-middle">
      <thead data-v-48717bcb="" class="sticky-top">
        <!-- <thead class=""> -->
        <tr data-v-48717bcb="">
          <th data-v-48717bcb="">ID <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Impact <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Created <i data-v-48717bcb="" class="bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Title <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">State <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Owner <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
        </tr>
      </thead>
      <tbody data-v-48717bcb="" class="table-group-divider">
        <tr data-v-15d9c71d="" class="osim-issue-queue-item osim-shaded">
          <td data-v-15d9c71d="" class="osim-issue-title"><a data-v-15d9c71d="" href="/flaws/CVE-2903-0092" class="">CVE-2903-0092</a></td>
          <td data-v-15d9c71d="" class="">MODERATE</td>
          <td data-v-15d9c71d="" class="">2021-07-29</td>
          <td data-v-15d9c71d="" class="">title</td>
          <td data-v-15d9c71d="" class="">NEW</td>
          <td data-v-15d9c71d="" class="">test@redhat.com</td>
          <!--<td>{{ issue.assigned }}</td>-->
        </tr>
        <!--v-if-->
      </tbody>
    </table><button data-v-48717bcb="" class="btn btn-primary" type="button">
      <!--v-if--><span data-v-48717bcb=""> Load More Flaws </span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`issueQueue > should render flaw labels 1`] = `
"<div data-v-48717bcb="" class="osim-content container-fluid osim-issue-queue">
  <div data-v-48717bcb="" class="osim-incident-filter">
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">My Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Open Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Hide labels</span></label></div>
    <!--v-if--><span data-v-48717bcb="" class="float-end"> Loaded 1 of 10</span>
  </div>
  <div data-v-48717bcb="" class="osim-incident-list">
    <table data-v-48717bcb="" class="table align-middle">
      <thead data-v-48717bcb="" class="sticky-top">
        <!-- <thead class=""> -->
        <tr data-v-48717bcb="">
          <th data-v-48717bcb="">ID <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Impact <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Created <i data-v-48717bcb="" class="bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Title <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">State <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Owner <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
        </tr>
      </thead>
      <tbody data-v-48717bcb="" class="table-group-divider">
        <tr data-v-15d9c71d="" class="osim-issue-queue-item osim-shaded">
          <td data-v-15d9c71d="" class="osim-issue-title pb-0"><a data-v-15d9c71d="" href="/flaws/CVE-2903-0092" class="">CVE-2903-0092</a></td>
          <td data-v-15d9c71d="" class="pb-0">MODERATE</td>
          <td data-v-15d9c71d="" class="pb-0">2021-07-29</td>
          <td data-v-15d9c71d="" class="pb-0">title</td>
          <td data-v-15d9c71d="" class="pb-0">NEW</td>
          <td data-v-15d9c71d="" class="pb-0">test@redhat.com</td>
          <!--<td>{{ issue.assigned }}</td>-->
        </tr>
        <tr data-v-15d9c71d="" class="osim-badge-lane osim-shaded">
          <td data-v-15d9c71d="" colspan="1">
            <div data-v-15d9c71d="" class="gap-1 d-flex flex-wrap">
              <!--v-if--><span data-v-15d9c71d="" style="background-color: rgb(255, 255, 255);" class="badge rounded-pill border text-bg-warning fw-bold border-warning text-decoration-line-through text-bg-gray border-secondary" title="Requested">test-3</span><span data-v-15d9c71d="" style="background-color: rgb(255, 255, 255);" class="badge rounded-pill border text-decoration-line-through text-bg-gray border-secondary" title="">test</span><span data-v-15d9c71d="" style="background-color: rgb(255, 255, 255);" class="badge rounded-pill border text-decoration-line-through text-bg-gray border-secondary" title="">test-2</span>
            </div>
          </td>
          <td data-v-15d9c71d="" colspan="90%"></td>
        </tr>
      </tbody>
    </table><button data-v-48717bcb="" class="btn btn-primary" type="button">
      <!--v-if--><span data-v-48717bcb=""> Load More Flaws </span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
